homeassistant/components/enocean/binary_sensor.py

```diff
@@ -1,56 +1,123 @@
 """Support for EnOcean binary sensors."""
 import logging
+from typing import List, Optional
 
 import voluptuous as vol
 
-from homeassistant.components import enocean
 from homeassistant.components.binary_sensor import (
     DEVICE_CLASSES_SCHEMA,
     PLATFORM_SCHEMA,
     BinarySensorEntity,
 )
 from homeassistant.const import CONF_DEVICE_CLASS, CONF_ID, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry
 import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_registry import async_entries_for_config_entry
+
+from .const import DATA_ENOCEAN, DOMAIN
+from .device import EnOceanEntity
+from .utilities import enocean_id_to_string, string_to_enocean_id
 
 _LOGGER = logging.getLogger(__name__)
 
 DEFAULT_NAME = "EnOcean binary sensor"
 DEPENDENCIES = ["enocean"]
 EVENT_BUTTON_PRESSED = "button_pressed"
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_ID): vol.All(cv.ensure_list, [vol.Coerce(int)]),
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
-    }
+PLATFORM_SCHEMA = vol.All(
+    cv.deprecated(CONF_ID),
+    PLATFORM_SCHEMA.extend(
+        {
+            vol.Required(CONF_ID): vol.All(cv.ensure_list, [vol.Coerce(int)]),
+            vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+            vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
+        },
+    ),
 )
 
+YAML_CONFIG_ENTRIES_TO_IMPORT = "config_entries_to_import"
+
+
+async def async_setup_platform(hass, config, add_entities, discovery_info=None):
+    """Convert the config entry to an entity in the registry."""
+    if not hass.config_entries.async_entries(domain=DOMAIN):
+        return
+
+    yaml_entries_to_migrate = hass.data[DATA_ENOCEAN].setdefault(
+        YAML_CONFIG_ENTRIES_TO_IMPORT, []
+    )
+    yaml_entries_to_migrate.append(config)
+
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Binary Sensor platform for EnOcean."""
-    dev_id = config.get(CONF_ID)
-    dev_name = config.get(CONF_NAME)
-    device_class = config.get(CONF_DEVICE_CLASS)
+async def async_setup_entry(hass: HomeAssistant, config_entry, async_add_entities):
```

Comment:
If there's no discovery of new devices and only manual yaml setup, we can remove `async_setup_entry` and only keep `setup_platform`.

Reply:
async_setup_entry method removed.